specifyweb/frontend/js_src/lib/components/SchemaConfig/UniquenessRules.tsx

```diff
@@ -0,0 +1,674 @@
+                    ([fieldName, _]) => fieldName
+                  ),
+                ];
+
+                const rows = fetchedDuplicates.fields.map(
+                  ({ duplicates, fields }) => [
+                    duplicates.toString(),
+                    ...Object.entries(fields).map(([_, fieldValue]) =>
+                      fieldValue.toString()
+                    ),
+                  ]
+                );
+
+                downloadDataSet(fileName, rows, columns, separator).catch(
+                  raise
+                );
+              }}
+            >
+              {schemaText.exportDuplicates()}
+            </Button.Danger>
+          ) : (
+            <Button.DialogClose>{commonText.close()}</Button.DialogClose>
+          )}
+        </>
+      }
+      dimensionsKey="ModifyUniquenessRule"
+      header={schemaText.configureUniquenessRule()}
+      icon={icons.pencilAt}
+      modal
+      onClose={handleClose}
+    >
+      <>
+        <H2>{label}</H2>
+        <p>{schemaText.uniqueFields()}</p>
+        {rule.fields.map((field, index) => (
+          <div className="inline-flex" key={index}>
+            <PickList
+              disabled={readOnly}
+              groups={{
+                [schemaText.fields()]: uniqueFields,
+              }}
+              value={field}
+              onChange={(value): void => {
+                const newField =
+                  fields.find(({ name }) => name === value) ??
+                  relationships.find(({ name }) => name === value);
+                if (newField === undefined) return;
+                handleChanged({
+                  ...rule,
+                  fields: replaceItem(rule.fields, index, newField.name),
+                });
+              }}
+            />
+            {rule.fields.length > 1 && (
+              <Button.Icon
+                className={`w-fit ${className.dataEntryRemove}`}
+                disabled={readOnly}
+                icon="minus"
+                title={commonText.remove()}
+                onClick={(): void =>
+                  handleChanged({
+                    ...rule,
+                    fields: removeItem(rule.fields, index),
+                  })
+                }
+              />
+            )}
+          </div>
+        ))}
+        <Button.BorderedGray
+          className="w-fit"
+          disabled={readOnly}
+          onClick={(): void =>
+            handleChanged({
+              ...rule,
+              fields: insertItem(
+                rule.fields,
+                rule.fields.length,
+                fields.find(({ name }) => !rule.fields.includes(name))!.name
+              ),
+            })
+          }
+        >
+          {commonText.add()}
+        </Button.BorderedGray>
+        <p>{schemaText.scope()}</p>
+        <UniquenessRuleScope
+          model={model}
+          rule={rule}
+          onChange={handleChanged}
+        />
+        <Input.Text
+          disabled
```

Comment:
did you mean `readonly` instead of disabled?
more semantically correct, and readonly is keyboard focusable